graphix/pattern.py

```diff
@@ -194,14 +194,14 @@ def __repr__(self):
             f"graphix.pattern.Pattern object with {len(self.__seq)} commands and {len(self.output_nodes)} output qubits"
         )
 
-    def equal(self, other):
+    def equal(self, other: "Pattern"):
```

Comment:
If you use `from __future__ import annotations`, you may get rid of quotation marks around `Pattern`.